### PR TITLE
ssh: allow all additional ssh key types

### LINKED ISSUE #59429: [BUG] not all openssh public key authentication types are supported
**Description**
The current ssh modules and states do not allow all supported key types for ssh public key authentication.

Any ssh pubkey type other than
- r
- rsa
- ssh-rsa
- e
- ecdsa
- ecdsa-sha2-nistp521
- ecdsa-sha2-nistp384
- ecdsa-sha2-nistp256
- ed25519
- ssh-ed25519

in this case `ssh-ed25519-cert-v01@openssh.com` fails with the following message:

```
          ID: ssh pubkey [...] for user root type ssh-ed25519-cert-v01@openssh.com
    Function: ssh_auth.present
        Name: [...]
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python3/dist-packages/salt/state.py", line 1919, in call
                  **cdata['kwargs'])
                File "/usr/lib/python3/dist-packages/salt/loader.py", line 1918, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/states/ssh_auth.py", line 349, in present
                  fingerprint_hash_type=fingerprint_hash_type)
                File "/usr/lib/python3/dist-packages/salt/modules/ssh.py", line 717, in set_auth_key
                  enc = _refine_enc(enc)
                File "/usr/lib/python3/dist-packages/salt/modules/ssh.py", line 78, in _refine_enc
                  'Incorrect encryption key type \'{0}\'.'.format(enc)
              salt.exceptions.CommandExecutionError: Incorrect encryption key type 'ssh-ed25519-cert-v01@openssh.com'.
     Started: 20:49:13.855666
    Duration: 1.456 ms
     Changes:   

```

This is a list of ssh pubkey types which are supported by openssh client version 8.4 for which the current implementation errors out (https://github.com/saltstack/salt/blob/master/salt/modules/ssh.py#L80):

```
ecdsa-sha2-nistp256-cert-v01@openssh.com
ecdsa-sha2-nistp384-cert-v01@openssh.com
ecdsa-sha2-nistp521-cert-v01@openssh.com
sk-ecdsa-sha2-nistp256-cert-v01@openssh.com
ssh-ed25519-cert-v01@openssh.com
sk-ssh-ed25519-cert-v01@openssh.com
rsa-sha2-512-cert-v01@openssh.com
rsa-sha2-256-cert-v01@openssh.com
ssh-rsa-cert-v01@openssh.com
sk-ecdsa-sha2-nistp256@openssh.com
sk-ssh-ed25519@openssh.com
rsa-sha2-512
rsa-sha2-256
```

**Versions Report**
I've checked several versions including the current master branch, all of which filter the key type and error out.

### PR DESCRIPTION
**Pull request by MEschenbacher**
### What does this PR do?

Propose a fix for #59429 by extending the list of allowed ssh public key types in `salt/modules/ssh.py` as of openssh 8.5.

### What issues does this PR fix or reference?
Fixes: #59429

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by xeacott** [comment]
Hey @MEschenbacher are you still wanting to make more changes to this PR? We can get this into Si just need to keep moving along

**Comment by MEschenbacher** [comment]
The PR should be complete as far as changes to code are concerned. Are you fine with the changes themselves?

I'm assuming in order to move along, I need to check all boxes?

File: salt/modules/ssh.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ def _refine_enc(enc):
         return enc
     elif enc in ed25519:
         return "ssh-ed25519"
-    else:
-        raise CommandExecutionError("Incorrect encryption key type '{}'.".format(enc))
+
```
Comment (Ch3LL) [inline salt/modules/ssh.py:68]: looks like @dwoz preferred option 1 from this comment https://github.com/saltstack/salt/issues/59429#issuecomment-794576844

@dwoz would you mind reviewing here.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
This will require a changelog file.

**Comment by MEschenbacher** [comment]
Okay, I'm now waiting for the final call as to whether we allow any pubkey type or continue maintaining an allowlist of types.

**Comment by xeacott** [comment]
I think the debian test is the last one to address here, and there's only one, and then we should be good to go!

**Review by dwoz — CHANGES_REQUESTED** [review]
Pre-commit and tests need to be fixed still.

**Comment by MEschenbacher** [comment]
I've added a few tests and I'm not sure if they assert that all salt modules and states with a ssh key type accept the additional key types. As far as the failing test for debian jinja filters is concerned: need to look into it.

**Comment by Ch3LL** [comment]
The jinja filters tests is a know failing test that someone else is working on and is not related to this PR. The pre-commit and lint test failures will need to be cleaned up though.

**Comment by Ch3LL** [comment]
I went ahead and pushed a fix for pre-commit and lint. Just waiting on the test results.

**Comment by Ch3LL** [comment]
@MEschenbacher is this still WIP? Or is it ready for review and merge?

**Comment by MEschenbacher** [comment]
Ready for review and merge.

**Review by Ch3LL — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**